src/system_info/system_info_utils.cc

```diff
@@ -48,7 +48,99 @@ char* GetDuidProperty() {
   }
   return s;
 }
-#endif
+
+#ifndef TIZEN_MOBILE
+GDBusConnection* GetDbusConnection() {
+  GError* error = NULL;
+  gchar* addr = g_dbus_address_get_for_bus_sync(G_BUS_TYPE_SYSTEM,
+                                                NULL, &error);
+  if (!addr) {
+    std::cout << "fail to get dbus addr: " << error->message << std::endl;
+    g_free(error);
+    return NULL;
+  }
+
+  GDBusConnection* bus_conn = g_dbus_connection_new_for_address_sync(addr,
+      (GDBusConnectionFlags)(G_DBUS_CONNECTION_FLAGS_AUTHENTICATION_CLIENT |
+      G_DBUS_CONNECTION_FLAGS_MESSAGE_BUS_CONNECTION), NULL, NULL, &error);
+  if (!bus_conn) {
+    std::cout << "fail to create dbus connection: "
+              << error->message << std::endl;
+    g_free(error);
+    return NULL;
+  }
+  return bus_conn;
+}
+
+char* OfonoGetModemPath(GDBusConnection* bus_conn) {
```

Comment:
Lease this to @zolkis. I am not familiar with how to get the active modem path. Please take a look